core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -3712,6 +3712,219 @@ protected PaginatedSearchResult doListUsers(String filter, int limit, int offset
         }
         return result;
     }
+    /**
+     * Count users in user stores.
+     *
+     * @param filter the filter for the user name. Use '*' to have all.
+     * @return user count
+     * @throws UserStoreException UserStoreException
+     */
+    public Long doCountUsers(String filter) throws UserStoreException {
+
+        Long usersCount = null;
+        Connection dbConnection = null;
+        String sqlStmt = null;
+        PreparedStatement prepStmt = null;
+        ResultSet resultSets = null;
+
+        try {
+
+            if (filter != null && filter.trim().length() != 0) {
+                filter = filter.trim();
+                filter = filter.replace("*", "%");
+            } else {
+                filter = "%";
+            }
+
+            dbConnection = getDBConnection();
```

Comment:
use try with resource for all the database connections, prepared statements and result sets.